lib/milc_interface.cpp

```diff
@@ -318,67 +318,83 @@ void qudaComputeOprod(int prec, int num_terms, int num_naik_terms, double** coef
+  qudaGaugeParam.mom_offset = arg->mom_offset;
+  qudaGaugeParam.site_size = arg->size;
+  qudaGaugeParam.gauge_order = arg->site ? QUDA_MILC_SITE_GAUGE_ORDER : QUDA_MILC_GAUGE_ORDER;
 
+  qudaGaugeParam.staggered_phase_applied = phase_in;
+  qudaGaugeParam.staggered_phase_type = QUDA_STAGGERED_PHASE_MILC;
+  qudaGaugeParam.t_boundary    = QUDA_ANTI_PERIODIC_T;
   if (!invalidate_quda_mom) {
-    gaugeParam.use_resident_mom = true;
-    gaugeParam.make_resident_mom = true;
+    qudaGaugeParam.use_resident_mom = true;
+    qudaGaugeParam.make_resident_mom = true;
   } else {
-    gaugeParam.use_resident_mom = false;
-    gaugeParam.make_resident_mom = false;
+    qudaGaugeParam.use_resident_mom = false;
+    qudaGaugeParam.make_resident_mom = false;
   }
 
-  updateGaugeFieldQuda(gauge, mom, eps, 0, 0, &gaugeParam);
+  updateGaugeFieldQuda(gauge, mom, eps, 0, 0, &qudaGaugeParam);
   qudamilc_called<false>(__func__);
   return;
 }
 
+void qudaUpdateU(int prec, double eps, QudaMILCSiteArg_t *arg)
+{
+  qudaUpdateUPhased(prec, eps, arg, 0);
+}
+
 void qudaRephase(int prec, void *gauge, int flag, double i_mu)
 {
   qudamilc_called<true>(__func__);
-  QudaGaugeParam gaugeParam = newMILCGaugeParam(localDim,
+  QudaGaugeParam qudaGaugeParam = newMILCGaugeParam(localDim,
       (prec==1) ? QUDA_SINGLE_PRECISION : QUDA_DOUBLE_PRECISION,
 						QUDA_GENERAL_LINKS);
 
-  gaugeParam.staggered_phase_applied = 1-flag;
-  gaugeParam.staggered_phase_type = QUDA_STAGGERED_PHASE_MILC;
-  gaugeParam.i_mu = i_mu;
-  gaugeParam.t_boundary    = QUDA_ANTI_PERIODIC_T;
+  qudaGaugeParam.staggered_phase_applied = 1-flag;
+  qudaGaugeParam.staggered_phase_type = QUDA_STAGGERED_PHASE_MILC;
+  qudaGaugeParam.i_mu = i_mu;
+  qudaGaugeParam.t_boundary    = QUDA_ANTI_PERIODIC_T;
 
-  staggeredPhaseQuda(gauge, &gaugeParam);
+  staggeredPhaseQuda(gauge, &qudaGaugeParam);
   qudamilc_called<false>(__func__);
   return;
 }
 
-void qudaUnitarizeSU3(int prec, double tol, QudaMILCSiteArg_t *arg)
+void qudaUnitarizeSU3Phased(int prec, double tol, QudaMILCSiteArg_t *arg, int phase_in)
 {
   qudamilc_called<true>(__func__);
-  QudaGaugeParam gaugeParam = newMILCGaugeParam(localDim,
+  QudaGaugeParam qudaGaugeParam = newMILCGaugeParam(localDim,
       (prec==1) ? QUDA_SINGLE_PRECISION : QUDA_DOUBLE_PRECISION,
 						QUDA_GENERAL_LINKS);
 
   void *gauge = arg->site ? arg->site : arg->link;
-  gaugeParam.gauge_offset = arg->link_offset;
-  gaugeParam.site_size = arg->size;
-  gaugeParam.gauge_order = arg->site ? QUDA_MILC_SITE_GAUGE_ORDER : QUDA_MILC_GAUGE_ORDER;
-
-  projectSU3Quda(gauge, tol, &gaugeParam);
+  qudaGaugeParam.gauge_offset = arg->link_offset;
+  qudaGaugeParam.site_size = arg->size;
+  qudaGaugeParam.gauge_order = arg->site ? QUDA_MILC_SITE_GAUGE_ORDER : QUDA_MILC_GAUGE_ORDER;
+  qudaGaugeParam.staggered_phase_applied = phase_in;
+  qudaGaugeParam.staggered_phase_type = QUDA_STAGGERED_PHASE_MILC;
+  qudaGaugeParam.t_boundary    = QUDA_ANTI_PERIODIC_T;
```

Comment:
Same comment as below, can we always assume this?

Reply:
Interpreting the comments from Carleton yes.